pkg/config/config.go

```diff
@@ -61,22 +59,22 @@ type IngressConfig struct {
 type MicroshiftConfig struct {
 	LogVLevel int `json:"logVLevel"`
 
-	SubjectAltNames []string `json:"subjectAltNames"`
-	NodeName        string   `json:"nodeName"`
-	NodeIP          string   `json:"nodeIP"`
-
-	Cluster ClusterConfig `json:"cluster"`
+	SubjectAltNames []string      `json:"subjectAltNames"`
+	NodeName        string        `json:"nodeName"`
+	NodeIP          string        `json:"nodeIP"`
+	BaseDomain      string        `json:"baseDomain"`
+	Cluster         ClusterConfig `json:"cluster"`
 
 	Ingress IngressConfig `json:"-"`
 }
 
-// Top level config
+// Top level config file
 type Config struct {
 	NodeName        string    `json:"nodeName"`
 	NodeIP          string    `json:"nodeIP"`
 	URL             string    `json:"url"`
-	ClusterDomain   string    `json:"clusterDomain"`
 	Network         Network   `json:"network"`
+	DNS             DNS       `json:"dns"`
```

Comment:
Can we call this `cluster`, please? The `baseDomain` (and possibly also the `clusterName` or whatever we call it) are properties of the cluster itself and used in more places than just DNS (which suggests CoreDNS), e.g. Ingress.

Reply:
That was part of the alignment @dinhxuanvu tried to do regarding openshift-install config, so I added it as-is. I would need him to weigh in. To me, your suggestion makes sense but would like to hear his opinion.

Reply:
Ok, I'd suggest to get this merged and do a follow-up PR, as this is not immediately related.